text/draw_text_shaper.cpp

```diff
@@ -49,11 +49,31 @@ class AdapterBuilder : public TextBlob::RunHandler {
   void commitRunBuffer(TextBlob::RunInfo& info) override
   {
     if (info.clusters && info.glyphCount > 0) {
-      float advanceX = 0.0f;
+      gfx::RectF selectionBounds;
+      gfx::RectF textBounds;
+      std::vector<gfx::RectF> glyphsBounds;
+      for (int i = 0; i < info.glyphCount; ++i) {
+        auto bounds = info.getGlyphBounds(i);
+        glyphsBounds.push_back(bounds);
```

Comment:
We can prealloc the expected size of `glyphsBounds`:
```suggestion
      std::vector<gfx::RectF> glyphsBounds(info.glyphCount);
      for (int i = 0; i < info.glyphCount; ++i) {
        auto bounds = info.getGlyphBounds(i);
        glyphsBounds[i] = bounds;
```